app/models/user.rb

```diff
@@ -738,8 +738,17 @@ def self.system_avatar_template(username)
 
   def self.letter_avatar_color(username)
     username ||= ""
-    color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
-    color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    if SiteSetting.restrict_letter_avatar_colors.present?
+      colors = SiteSetting.restrict_letter_avatar_colors.split("|")
```

Comment:
I think we should avoid the splitting here since we already know the value in site settings is valid. 

So.... `aabbcc|ccddee` has 2 colors cause it is (6+6+1) long

Then we decide on an index ... and do a substring from index * 6 + 1 for length 6. 

This means we don't have to keep splitting the string over and over for every user on the page.